AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/IncomingAsyncSubscriptionEventPublisher.swift

```diff
@@ -147,12 +150,21 @@ final class IncomingAsyncSubscriptionEventPublisher: Cancellable {
         for modelType: Model.Type,
         subscriptionType: GraphQLSubscriptionType,
         api: APICategoryGraphQLBehavior,
+        auth: AuthCategoryBehavior?,
         valueListener: @escaping GraphQLSubscriptionOperation<Payload>.InProcessListener,
         completionListener: @escaping GraphQLSubscriptionOperation<Payload>.ResultListener
     ) -> GraphQLSubscriptionOperation<Payload> {
 
-        let request = GraphQLRequest<Payload>.subscription(to: modelType,
-                                                           subscriptionType: subscriptionType)
+        let request: GraphQLRequest<Payload>
+        if let auth = auth, modelType.schema.hasAuthenticationRules, let user = auth.getCurrentUser() {
+            // TODO: check model schema for identityClaim to figure out which is the ownerId field coming from
```

Comment:
Seems like we should open an issue around this.

Reply:
good idea, adding issue https://github.com/aws-amplify/amplify-ios/issues/485, impact, and linked in comment